<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
  <meta name="description" content="">
  <meta name="author" content="">
  <link rel="icon" type="image/png" href="images/favicon-32x32.png" sizes="32x32" />
  <link rel="icon" type="image/png" href="images/favicon-16x16.png" sizes="16x16" />

  <title>Bioinformatics Project</title>
  <!--stylesheet-->


  <link href="https://fonts.googleapis.com/css?family=Source+Sans+Pro:400,700,900" rel="stylesheet">
  <link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet" type="text/css">
  <link href="styles/styles.css" rel="stylesheet" type="text/css">
  <link href="styles/custom-responsive-styles.css" rel="stylesheet" type="text/css">

  <!--scripts-->
  <script type="text/javascript" src="scripts/jquery-3.2.1.min.js"></script>
  <script type="text/javascript" src="scripts/all-plugins.js"></script>
  <script type="text/javascript" src="scripts/plugins-activate.js"></script>

  <style>
  table{

        border: solid 2px black;
  }

  tr, td, th{
    border: solid 2px black;
  }
  </style>

</head>

<body id="page-top">
  <!-- Navigation -->

  <a class="menu-toggle rounded" href="#">
    <i class="fa fa-bars"></i>
  </a>
  <nav id="sidebar-wrapper">
    <ul class="sidebar-nav">
      <li class="sidebar-brand">
        <a class="smooth-scroll" href="#Header"></a>
      </li>
      <li class="sidebar-nav-item">
        <a class="smooth-scroll" href="#page-top">Home</a>
      </li>
      <li class="sidebar-nav-item">
        <a class="smooth-scroll" href="#About">About</a>
      </li>
      <li class="sidebar-nav-item">
        <a class="smooth-scroll" href="#ProjectProgress">Project Progress</a>
      </li>
      <li class="sidebar-nav-item">
        <a class="smooth-scroll" href="#ProjectProposal">Project Proposal</a>
      </li>

    </ul>
  </nav>
  <!-- Header Starts -->
  <section id="Banner" class="content-section">
    <div class="container content-wrap text-center">
      <p style="font-weight: bold;font-size: 2.3em;">Finding a Machine Learning Model that Best Predicts Canola Traits.</p>
      <h3 >
          <em>COMP 4520 Honors Project</em>
        </h3>
    </div>
    <div class="overlay"></div>
  </section>
  <!-- Header Ends -->
  <!-- About Us Starts -->
  <section id="About" class="content-section">
    <div class="container text-center">
      <div class="row">
        <div class="col-lg-12">
          <div class="block-heading">
            <h2>About</h2>
          </div>
          <p class="lead">Canola is proudly grown in Canada and is now used as a source of oil. At the Faculty of Agriculture, University of Manitoba, Scientists have planted about 460 Canola seeds each with a different Genetic Sequence. They then recorded the Genetic Sequence of each seed and its corresponding traits. We will use this recorded data to build Machine learning models and find the best Model that will in turn be used to predict Canola crop traits. As a result, researchers can breed new varieties of Canola and easily predict and understand their traits without having to plant them. Furthermore, using these predictions, scientists can quickly find the Genetic Sequence that gives the most favourable traits and thus those seeds can be used by Canola farmers to maximize their revenue. </p>
        </div>
      </div>
    </div>
  </section>
  <!-- About Us Starts -->
  <section id="ProjectProgress" class="content-section text-center">
    <div class="container">
      <div class="block-heading">
        <h2>Project Progress</h2>
        <div>
          <table>

            <tr>
              <th style="width: 110%;">Work Done</th>
              <th>Milestone #</th>
            </tr>

            <tr>
              <td><p> Build some Machine Learning Models using XGboost and SVM. There were some promising results and meanwhile we are working on mapping the most influenced one hot encoded features to the original features </p></td>
              <td><p> 1 (June 14th)</p></td>
            </tr>
            
            <tr>
              <td>
                <p>We ran Models with One Hot Encoding and without One Hot Encoding. There appears to be a noticeable differnce and I am working on re-running it several times
                to get a little more confidence that exploring the one hot encoding part is the right way to go. I will try to run the model as many times as I can, usually takes about 8 hours for a model to train and sometimes it crashes. By next week I will send Mike any results that I receive.</p>
                  
                <p>In addition to that, I and Mike decided to build a mini program that helps map features from One hot encoding to 
                actual original genomes

                
                Make a program that verifies that there is at least one -1, 0, 1 in each column of the dataset.
                If not all columns in the dataset have at least one -1, 0, 1, then I will modify the program to ensure that it manually produces a CSV file that contains headers (gene-identifier) in every column and 
                  the rows underneath each header will have the actual column number that would be produced after one hot encoding. This means that once we have the column numbers of the most influential markers, we can easyly use this
                  CSV file that my program produces to look up the genetic marker for that column.
                  
                  We will then run models on Yield and find the most influential markers. We will also look for overlappings over different runs.</p>
               
                <p>Please note: Due to a single taking about 8 hours to run, the above plan will take weeks to accomplish</p>
                
              </td>
              <td><p > 2 (June 25th)</p></td>
            </tr>
            
            <tr>
              <td><p>I made a program that mapped One Hot Encoding mappings back to the original genomes. This program generates a CSV file and the column headers are the genetic markings. The data under the column headers of each
                genetic marking has the column number of the row the respective genetic marking is mapped in the One Hot. This was a Java program and takes in some pre-processed data. The reason I did this was because there are some rows
                in the dataset that dont have at least one -1,0,1 in them. Moving forward, we decided to focus on making working code for Flowering time and doing multiple runs. In the process we will attempt
                to extract the most influenced Genetic Markers for flowering time. We can them use the same code and apply it to Yield later on.
                </p></td>
              <td><p>3 (July 2nd)</p></td>
            </tr>
            
             <tr>
              <td><p>I coded the loop that is finds the genetic markers that most influence flowering time. I ran the code on local machine but the run seems to freeze and I also tried to run 
                it on Aviary but the same happens there. I used most of my time during this week for coding the loop and learning how to set up runs on Aviary. After each loop cycle, we have a list
                of genetic markers that influence flowering time. These results are stored in a dictionary and it loop again and records any new markers present and any overlappings. Currently
                the model only runs once, but I and Mike decided that the model should run as many times as the loop runs. I will modify the code and do trials, that is the direction we decided to go</p></td>
              <td><p>4 (July 9th)</p></td>
            </tr>
            
            <tr>
              <td><p>I worked on running various machine learning models that involved flowering time and Yield. These models were trained 100 times each and the results printed out.
                I also worked on structuring the data to get meaningful results which involed the marker positions and chromosomes that Mike sent me. There appears to be some wierd 
                behaviour with both flowering time and Yield when we look at which chromosomes most influence these respective traits. The weird behaviour is that we are only getting the starting
                chromosomes and none other after that. This is in contrast to our expectations that the results should be spread out throughout the chromosomes. For this reason, I will be running models after 
                changing the values of K in feature selection, getting rid of feature selection altogether and using all the markers instead of filtering the ones that appear > 10. I will do
                various experimentation around that to see if its giving us the same results and if we are able to get rid of the wierd behaviour.</p></td>
              <td><p>Comming Soon!</p></td>
            </tr>
            
             <tr>
              <td><p style="color:red" >Comming Soon!</p></td>
              <td><p style="color:red" >Comming Soon!</p></td>
            </tr>

          </table>
        </div>

      </div>

  </section>
  <section class="content-section text-center" id="ProjectProposal">
    <div class="container">
      <div class="block-heading">
        <h2 style="margin-bottom:30px">Project Proposal </h2>
        <iframe src="Template-UGradHonours.html" style="width: -webkit-fill-available;
    height: 400px;    border: solid 2px black;"></iframe><button type="submit" style="float:left" class="btn btn-primary btn-xl" onclick="window.open('UGradHonours.pdf')">Download</button>




      </div>



    </div>
  </section>

  <footer class="footer text-center">
    <div class="container">
      <ul class="list-inline">
        <li class="list-inline-item">
          <a class="social-link rounded-circle text-white" href="mailto:patelm5@myumanitoba.ca">
              <i class="fa fa-envelope" aria-hidden="true"></i>
            </a>
        </li>

        <li class="list-inline-item">
          <a class="social-link rounded-circle text-white" href="https://www.linkedin.com/in/mitul-patel-6a8853182/">
              <i class="fa fa-linkedin" aria-hidden="true"></i>
            </a>
        </li>

        <li class="list-inline-item">
          <a class="social-link rounded-circle text-white" href="https://github.com/Mitul264">
              <i class="fa fa-github" aria-hidden="true"></i>
            </a>
        </li>


      </ul>
      <p>Mitul Patel<br>Computer Science - Honors Student<br>University of Manitoba</p>

    </div>
  </footer>
</body>

</html>
